### PR TITLE
Improving ImplementInterface code fix

### PR DESCRIPTION
**Pull request by psfinaki**
Changes are small (review without whitespace) and include:
- move to ctasks
- removing some duplicated async calls
- tests
- half-assed fix of #5813: as in, the code fix just doesn't trigger in that case now - generating correct fixed code should be doable but hard and not worth my effort given it's a rare case

### CODE REVIEW

File: vsintegration/src/FSharp.Editor/CodeFixes/ImplementInterfaceCodeFixProvider.fs
Comment (psfinaki) [inline vsintegration/src/FSharp.Editor/CodeFixes/ImplementInterfaceCodeFixProvider.fs:1]: Rename to `ImplementInterface` prior to merging.

File: vsintegration/src/FSharp.Editor/CodeFixes/ImplementInterfaceCodeFixProvider.fs [hunk 1]
```diff
@@ -32,15 +30,19 @@ type internal InterfaceState =
         Tokens: Tokenizer.SavedTokenInfo[]
     }
 
-[<ExportCodeFixProvider(FSharpConstants.FSharpLanguageName, Name = "ImplementInterface"); Shared>]
+// state machine not statically compilable
+// TODO: rewrite token arithmetics properly here
```
Comment (abonie) [inline vsintegration/src/FSharp.Editor/CodeFixes/ImplementInterfaceCodeFixProvider.fs:34]: Is this TODO for later?

Comment (psfinaki) [reply vsintegration/src/FSharp.Editor/CodeFixes/ImplementInterfaceCodeFixProvider.fs:34]: Yeah. For those who like doing token arithmetics. There should be at least a few folks for that on this planet.

Comment (abonie) [reply vsintegration/src/FSharp.Editor/CodeFixes/ImplementInterfaceCodeFixProvider.fs:34]: Ok, I usually use TODO for stuff I mean to resolve within given PR in a future commit, hence the question

**Review by abonie — APPROVED** [review]

**Pull request merged**